dosbox-staging/dosbox-staging
# Centralize app_id handling

# Description

On modern Linux it is preferred to use the fully qualified application ID for naming the desktop, icon, and metainfo files. Add a meson option to capture this variable and set it to `org.dosbox-staging.DOSBoxStaging` which better matches that the homepage is dosbox-staging.org.

Then ensure that we're setting the X11 `resourceclass` and the Wayland app_id to this value. Those values need to match the name of the .desktop file in order for compositors/window managers to correctly associate windows to the appropriate .desktop file (see [here](https://github.com/zed-industries/zed/issues/12707#issuecomment-2168742761) for an explanation I previously wrote on how that works). Now that we're ensuring that both X11 and Wayland should have correct window associations we can drop `StartupWMClass` from the desktop file as it is now unnecessary.

Also, since SDL3 changed how that is done I added some #if statements so that that's ready whenever dosbox-staging is ported to SDL3.

# Manual testing
- Checked output of installing a meson build to a temporary directory to ensure that files were named correctly and had the correct values
- Launched built dosbox and verified that it had the correct X11 resourceclass

# Checklist

I have:
